src/utils/useMapboxSearch.ts

```diff
@@ -0,0 +1,72 @@
+import { useCallback, useMemo, useState } from 'react';
+
+import {
+  AddressAutofillCore,
+  AddressAutofillFeatureSuggestion,
+  AddressAutofillSuggestion,
+  SearchSession,
+} from '@mapbox/search-js-core';
+
+import strings from 'src/strings';
+
+import useMapboxToken from './useMapboxToken';
+import useSnackbar from './useSnackbar';
+
+// Wait a quarter second before calling suggest again
+const SUGGESTION_DELAY_MS = 250;
+
+export type MapboxSearch = {
+  clear: () => void;
+  retrieve: (suggestion: AddressAutofillSuggestion) => Promise<AddressAutofillFeatureSuggestion[]>;
+  suggest: (suggestText: string) => Promise<AddressAutofillSuggestion[]>;
+  suggestText?: string;
+};
+
+const useMapboxSearch = (): MapboxSearch => {
+  const { token } = useMapboxToken();
+
+  const addressAutofillCore = new AddressAutofillCore({
+    accessToken: token,
+  });
+  const session = new SearchSession(addressAutofillCore, SUGGESTION_DELAY_MS);
```

Comment:
Should this and the `session.addEventListener` be inside a `useEffect`? These will fire again any time a component using this hook rerenders, but that may be OK

Reply:
Good idea! Doesn't seem like it will hurt.